### PR TITLE
Update OpenTelemetry Collector documentation

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

Improves OpenTelemetry Collector documentaiton to reflect changes on the Collector, the OTLP protocol  and the Datadog exporter. More specifically:

- Updates OTLP/gRPC port to 4317
- Update information about hostname resolution in the Datadog exporter
- Remove requirements related to a timeout of 10 seconds in the exporter
- Add note about `latest` tag on Docker image
- Add `--hostname $(hostname)` option for Docker deployments to ensure the hostname is accurate

### Motivation

Ensure docs are up to date.

### Preview

See https://github.com/DataDog/documentation/pull/14542#issuecomment-1175914661

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/setup_overview/open_standards/otel_collector_datadog_exporter.md [hunk 1]
```diff
@@ -25,12 +25,12 @@ datadog:
     site: {{< region-param key="dd_site" code="true" >}}
 ```
 
-On each OpenTelemetry-instrumented application, set the resource attributes `deployment.environment`, `service.name`, and `service.version` using [the language's SDK][1]. As a fall-back, you can also configure environment, service name, and service version at the collector level for unified service tagging by following the [example configuration file][7]. The exporter attempts to get a hostname by checking the following sources in order, falling back to the next one if the current one is unavailable or invalid:
```
Comment (mx-psi) [inline content/en/tracing/setup_overview/open_standards/otel_collector_datadog_exporter.md:28]: These options existed, but they are deprecated and will be removed before GA

File: content/en/tracing/setup_overview/open_standards/otel_collector_datadog_exporter.md [hunk 2]
```diff
@@ -57,27 +53,22 @@ receivers:
 
 processors:
   batch:
-    timeout: 10s
-  resourcedetection:
-    detectors: [gce, ecs, ec2, azure, system]
```
Comment (mx-psi) [inline content/en/tracing/setup_overview/open_standards/otel_collector_datadog_exporter.md:62]: This is supported but not required; removed to simplify the example

File: content/en/tracing/setup_overview/open_standards/otel_collector_datadog_exporter.md [hunk 3]
```diff
@@ -98,28 +89,34 @@ service:
...
 
-3.  Determine which ports to open on your container. OpenTelemetry traces are sent to the OpenTelemetry Collector over TCP or UDP on several ports, which must be exposed on the container. By default, traces are sent over OTLP/gRPC on port `55680`, but common protocols and their ports include:
+3.  Determine which ports to open on your container. OpenTelemetry traces are sent to the OpenTelemetry Collector over TCP or UDP on several ports, which must be exposed on the container. By default, traces are sent over OTLP/gRPC on port `4317`, but common protocols and their ports include:
 
       - Zipkin/HTTP on port `9411`
       - Jaeger/gRPC on port `14250`
       - Jaeger/HTTP on port `14268`
       - Jaeger/Compact on port (UDP) `6831`
-      - OTLP/gRPC on port `55680`
+      - OTLP/gRPC on port `4317`
```
Comment (mx-psi) [inline content/en/tracing/setup_overview/open_standards/otel_collector_datadog_exporter.md:109]: The default port was changed to 4317 (a long time ago!) https://github.com/open-telemetry/opentelemetry-specification/issues/1148#issuecomment-742080390

File: content/en/tracing/setup_overview/open_standards/otel_collector_datadog_exporter.md [hunk 4]
```diff
@@ -25,12 +25,12 @@ datadog:
     site: {{< region-param key="dd_site" code="true" >}}
 ```
 
-On each OpenTelemetry-instrumented application, set the resource attributes `deployment.environment`, `service.name`, and `service.version` using [the language's SDK][1]. As a fall-back, you can also configure environment, service name, and service version at the collector level for unified service tagging by following the [example configuration file][7]. The exporter attempts to get a hostname by checking the following sources in order, falling back to the next one if the current one is unavailable or invalid:
+On each OpenTelemetry-instrumented application, set the resource attributes `deployment.environment`, `service.name`, and `service.version` using [the language's SDK][1]. The exporter attempts to get a hostname by checking the following sources in order, falling back to the next one if the current one is unavailable or invalid:
```
Comment (gbbr) [inline content/en/tracing/setup_overview/open_standards/otel_collector_datadog_exporter.md:28]: I feel like these two sentences should not be in the same paragraph.

**Review by gbbr — COMMENTED** [review]
This LGTM, I'm going to try and update the K8s docs soon.

File: content/en/tracing/setup_overview/open_standards/otel_collector_datadog_exporter.md (hunk 4 above)
Comment (mx-psi) [reply content/en/tracing/setup_overview/open_standards/otel_collector_datadog_exporter.md]: Agree, fixed

**Review by buraizu — APPROVED** [review]
Looks good from a docs perspective, but what's the reason for changing all the numbers to 1?

**Comment by mx-psi** [comment]
> Looks good from a docs perspective, but what's the reason for changing all the numbers to 1?

It was just so that I didn't have to update them every time; Markdown will replace it with the actual numbers. Happy to revert that change if it goes against conventions in this repository :)

**Comment by mx-psi** [comment]
@buraizu could you merge this?

File: content/en/metrics/otlp.md [hunk 5]
```diff
@@ -121,14 +121,13 @@ You may add all resource attributes as tags by using the `resource_attributes_as
 OpenTelemetry defines certain semantic conventions related to host names. If an OTLP payload has a known hostname attribute, Datadog honors these conventions and tries to use its value as a hostname. The semantic conventions are considered in the following order:
 
 1. `datadog.host.name`, a Datadog-specific hostname convention
-2. `k8s.node.name`, the Kubernetes node name
-3. Cloud provider-specific conventions, based on the `cloud.provider` semantic convention
-4. `host.id`, the unique host ID
-5. `host.name`, the system hostname
-6. `container.id`, the container ID
+1. Cloud provider-specific conventions, based on the `cloud.provider` semantic convention
+1. Kubernetes-specific conventions from the `k8s.node.name` and `k8s.cluster.name` semantic conventions.
```
Comment (bkabrda) [inline content/en/metrics/otlp.md:125]: None of the other points in this list end with a `.`, so maybe this one shouldn't as well?

**Review by bkabrda — APPROVED** [review]
LGTM, just one nit comment left inline.

File: content/en/metrics/otlp.md (hunk 5 above)
Comment (mx-psi) [reply content/en/metrics/otlp.md]: Makes sense, removed

**Pull request merged**